risks/MASVS-STORAGE/1-store-sensitive-data-securely/data-unencrypted-internal/android-data-in-sandbox/test.md

```diff
@@ -0,0 +1,30 @@
+---
+platform: android
+title: Sensitive Data Written to Private Data Directory (Sandbox) Unencrypted.
+type: [dynamic, filesystem]
+mitigations:
+- android-use-keystore
+- android-use-androidx-security
+prerequisites:
+- identify-sensitive-data
+---
+
+## Steps
+
+1. Start the device.
+
+2. Launch and use the app going through the various workflows while inputting sensitive data wherever you can. Taking note of the data you input can help identify it later using tools to search for it.
+
+3. Take a copy of the app's private data directory for offline analysis. Using tar will preserve the filesystem structure and permissions.
```

Comment:
I wonder if the step 3 should be more detailed, especially for users who are familiar with those type of taks or techniques.

- How would they extract the data? 
- What tool would they used? 
- Where could they find some documentation about this?

Reply:
My understanding is that the detailed steps and tools would be in the examples under this test, and this just describes the high level process. The examples can then show more than one way or tool. Probably a good questions to clarify in the call next week.

Reply:
Agreed on the call that the more specific steps would be in the example tests.